Adicionando 1273 (#153)
petbccufscar/URI: Strings/1273/README.md

# Problema

Nós temos algumas palavras e queremos justificá-las à direita, ou seja, alinhar todas elas à direita. Crie um programa que, após ler várias palavras, reimprima estas palavras com suas linhas justificadas à direita.

###### Problema completo: https://www.urionlinejudge.com.br/judge/pt/problems/view/1273

# Resolução

Para resolver esse problema, precisamos primeiro receber quantas palavras justificaremos e todas as palavras. Assim, conseguimos descobrir a maior palavra e quanto para a direita precisaremos afastar a palavra para justificá-la em relação a maior.

Usaremos a biblioteca `string.h` para utilizar a função `strlen(char* string)` que nos retorna o tamanho de uma dada string.

> 

Author: ```suggestion
Como faremos algumas comparações ente dois inteiros para definir o maior entre eles, criamos a função `int max(int a, int b)` que com um `if` nos retorna o maior entre os dois números nela definidos como `a` e `b`.

```c
int max(int a,int b){
    if(a>b)
        return a;
    else
        return b;
}
```
```

# Problema

Nós temos algumas palavras e queremos justificá-las à direita, ou seja, alinhar todas elas à direita. Crie um programa que, após ler várias palavras, reimprima estas palavras com suas linhas justificadas à direita.

> Entrada
> A entrada contém diversos casos de testes. A primeira linha de cada caso de teste conterá um inteiro N (1 ≤ N ≤ 50), que indicará o número de palavras que virão a seguir. Cada uma das N palavras contém no mínimo uma letra e no máximo 50 letras maiúsculas (‘A’-‘Z’). O fim da entrada é indicado por N = 0.

Author: ```suggestion

```

###### Problema completo: https://www.urionlinejudge.com.br/judge/pt/problems/view/1273

# Resolução

Para resolver esse problema, precisamos primeiro receber quantas palavras justificaremos e todas as palavras. Assim, conseguimos descobrir a maior palavra e quanto para a direita precisaremos afastar a palavra para justifica-la em relação a maior.

> Usaremos a biblioteca `string.h` para utilizar a função `strlen(str string)` que nos retorna o tamanho de uma dada string.

Author: ```suggestion
Usaremos a biblioteca `string.h` para utilizar a função `strlen(char* string)` que nos retorna o tamanho de uma dada string.
```

```c
#include <string.h>

```

Começaremos declarando as variáveis que iremos utilizar. `N` será quantas palavras receberemos (0 para parar a execução), `i,j e t` são auxiliares, `maior_palavra` é o tamanho da maior palavra recebida. A matriz `palavras[50][50]` é uma lista de palavras, que recebe até 50 palavras de tamanho 50. 

```c
    int N, i,maior_palavra = 0, j,t;
    char palavras[50][50];
```

Assim, recebemos o valor do primeiro `N` e iniciamos o loop para os próximos valores, ao fim do loop recebemos o próximo `N` para novamente passar pela verificação do `while`.

```c
    scanf("%d",&N);
    while(N != 0){

        scanf("%d",&N);
    }
```

> Dentro do loop, começamos zerando a variável `maior_palavra` que será diferente para cada conjunto de palavras recebida. Logo em seguida, criamos um loop para receber todas as `N` palavras e a cada palavra recebida, utilizamos  para substituir o conteúdo da variável `maior_palavra` pelo retorno da função `max()` que retorna o maior entre o tamanho da palavra atual e da `maior_palavra`

Reviewer: ```suggestion
Dentro do loop, começamos zerando a variável `maior_palavra`, que será diferente para cada conjunto de palavras recebida. Logo em seguida, criamos um loop para receber todas as `N` palavras e a cada palavra recebida, substituímos o conteúdo da variável `maior_palavra` pelo retorno da função `max()` que retorna o maior entre o tamanho da palavra atual e da `maior_palavra`.
```
Acho que é importante mencionar que você teve que programar a função `max()` no código.
